modules/node/src/channel/channel.service.ts

```diff
@@ -264,16 +262,23 @@ export class ChannelService {
   async getChannelState(userPublicIdentifier: string): Promise<any> {
     const channel = await this.channelRepository.findByUserPublicIdentifier(userPublicIdentifier);
     if (!channel) {
-      throw new Error(`No channel exists for userPublicIdentifier ${JSON.stringify(userPublicIdentifier)}`);
+      throw new Error(
+        `No channel exists for userPublicIdentifier ${userPublicIdentifier}`,
+      );
     }
     const { data: state } = await this.cfCoreService.getStateChannel(channel.multisigAddress);
 
-    // FIXME: this has to be [] not {}, we should fix this better
+    // FIXME: these need to be [] not {}, we should fix this better
     state.singleAssetTwoPartyIntermediaryAgreements =
       Object.values(state.singleAssetTwoPartyIntermediaryAgreements).length === 0
         ? []
         : state.singleAssetTwoPartyIntermediaryAgreements;
 
+    state.proposedAppInstances =
+      Object.values(state.proposedAppInstances).length === 0 ? [] : state.proposedAppInstances;
+
+    state.appInstances = Object.values(state.appInstances).length === 0 ? [] : state.appInstances;
+
```

Comment:
Cleaner fix included in #577

Reply:
Fix in 577 should prevent more bad values being saved to the db but this'll still damage-control the ones that are already there. Let's keep this until we can confirm the db no longer has any bad states.

Reply:
Wait, why?

The `.toJson` operates on functions being retrieved from the db, im not sure that it would prevent anything from being saved incorrectly (except on the client side, in which case restore state should fix it)